packages/medusa/src/repositories/gift-card.ts

```diff
@@ -64,12 +74,37 @@ export class GiftCardRepository extends Repository<GiftCard> {
             .orWhere(`display_id::varchar(255) ILIKE :dId`, { dId: `${q}` })
         })
       )
-      .getMany()
 
-    return this.findWithRelations(
+    let raw: GiftCard[] = []
+    let count = 0
+    if (shouldCount) {
+      const [results, resultCount] = await qb.getManyAndCount()
+      raw = results
+      count = resultCount
+    } else {
+      raw = await qb.getMany()
+    }
+
+    const [results] = await this.findWithRelations(
       rels,
       raw.map((i) => i.id)
     )
+
+    return [results, count]
+  }
+
+  public async listGiftCardsAndCount(
+    query: ExtendedFindConfig<GiftCard, QuerySelector<GiftCard>>,
+    rels: (keyof GiftCard | string)[] = [],
+    q?: string
+  ): Promise<[GiftCard[], number]> {
+    if (q) {
+      const where = query.where
```

Comment:
suggestion(nit non blocking at the moment): I would suggest that as a rule of thumb, when we do that we should spread the object into a new one. Here we are assigning the reference of that object and deleting the id, which will result in the fact that the consumer query has no id as well since it has been removed from the referenced object that they are sharing. It could be tricky to debug if the consumer want to reuse that query object elsewhere 😂 wdyt? an example in the screenshot where you can see that in the service the id is removed as well. I forced the example manually just to show the result

<img width="1496" alt="Screenshot 2022-07-02 at 18 33 46" src="https://user-images.githubusercontent.com/25098370/177008849-e08873d4-ad2b-4f05-9abb-716a90631319.png">

Reply:
Great point!